Create with-sveltekit.mdx

timsonner opened the pull request:
Add SvelteKit SupaBase implementation.

## What kind of change does this PR introduce?

Docs update

## What is the current behavior?

Svelte as the default SupaBase implementation

## What is the new behavior?

SvelteKit as an alternative implementation, updates to navigating the SupaBase website, Vite specific environmental variables, usage of $lib for imports and accessing components in the libs directory, abolishment of rollup, because not need, added examples of changing ports for the dev server.

## Additional context

Add any other context or screenshots.

timsonner committed:
Create with-sveltekit.mdx

Create SvelteKit guide.

kiwicopple commented:
This is awesome @timsonner - I will tag it as `help-wanted` in case someone from the community has a chance to review it. If it's still open in a few days then I can jump in and try it out :)

Isaiah-Hamilton reviewed (commented):
@timsonner You need to add a link to the docs in the sidebar so people can access it.

P.S. keep the sidebar in alphabetical order to not favor one framework over the other.

Isaiah-Hamilton commented on web/docs/guides/with-sveltekit.mdx:
```diff
@@ -0,0 +1,516 @@
+<script>
+	import {user} from "$lib/sessionStore"
+	import {supabase} from "$lib/supabaseClient"
+	import Auth from "$lib/Auth.svelte"
+	import Profile from "$lib/Profile.svelte"
```
These 2 imports seem to be broken. I think you need to remove the `.svelte` at the end

timsonner replied:
Ok, I just removed them

timsonner replied:
Hey @Isaiah-Hamilton, I just realized that removing '.svelte' doesn't fix it, I should have tested before the commit. I have no idea why this is hit and miss, but it actually won't work for me without the '.svelte'. I did hit ctrl-c and restart the dev server a few times, but it did give me the same SSR module load error at first, then it went away.  I'm not in a hurry to ship this, just trying to get to the bottom of it. This is kinda strange because, almost buggy. I have a working copy of the app here: https://github.com/timsonner/sveltekit-supabase. I'll have to go back through the guide and see what I missed, this may be a bug. Can you try ctrl-c and restarting the dev server a few times? It's not a great solution, but I can't figure out why this doesn't work, then does.

timsonner replied:
Sorry, also in Auth.svelte I changed the import to: "import {supabase} from '$lib/supabaseClient'" instead of "import {supabase} from './supabaseClient'" this may have been the problem and it was supposed to make it into a previous commit. Kinda just got all scrambled and out of sync with my commits and multiple project windows open.

Isaiah-Hamilton reviewed (commented):
Getting error when trying to start localhost.
<img width="500" alt="Screen Shot 2022-01-26 at 5 57 16 PM" src="https://user-images.githubusercontent.com/70828596/151261182-6cb3320d-331a-4b9c-be6c-7b8538046689.png">
<img width="600" alt="Screen Shot 2022-01-26 at 5 57 57 PM" src="https://user-images.githubusercontent.com/70828596/151261183-fe12b6d7-669d-4a81-9545-752d37b511f6.png">

timsonner committed:
Remove '.svelt' from imports

timsonner committed:
Update with-sveltekit.mdx

Added '.svelte' back, previous commit added '$lib', and removed './' (forgot) So $lib seems to be the fix, not removing '.svelte'. In fact, it won't work for me without '.svelte', I think the fix was using "import Auth from '$lib/Auth.svelte'" instead of "import Auth from './Auth.svelte'". Sorry so confusing, I forgot I nixed the relative path in the import and used $lib instead, this took me 45 min to figure out.

Isaiah-Hamilton reviewed (commented):
ok looks great just need to delete the `.gitpod.yml` and `package-lock.json` files, and add your link to the doc in the sidebar.

timsonner commented:
So sorry to be away so long, I will work on this soon.

timsonner committed:
Delete .gitpod.yml

Remove gitpod.yml

timsonner committed:
Delete package-lock.json

Remove package-lock.json

timsonner reviewed (commented):
Removed gitpod.yml and package-lock.json. Still need to add link in sidebar.

Isaiah-Hamilton committed:
chore: add back package-lock.json

Isaiah-Hamilton committed:
chore: update sveltekit docs

Isaiah-Hamilton committed:
chore: add sveltekit docs to sidebar
